### PR TITLE
Fix correctness of JAX fused attention with CP and improve numerics check in unit tests

### PR DESCRIPTION
**Pull request by mgoldfarb-nvidia**
# Description

Fix multiple JAX CP fused attention bugs masked by tiny gradients in the unit tests.

The unit tests are updated to use the same gradient scaling mechanism as the non-distributed fused attention tests.

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

## Changes

Bug fixes to CP fused attention for Jax and updated unit test.

# Checklist:

- [X] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [X] The functionality is complete
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by mgoldfarb-nvidia** [comment]
\te-ci jax

File: tests/jax/test_distributed_fused_attn.py [hunk 1]
```diff
@@ -510,37 +546,25 @@ def ref_func(q, k, v, mask, kv_groups):
...
 
-            try:
-                assert_allclose(target_fwd, ref_fwd, dtype=dtype)
-            except AssertionError as e:
-                has_diffs = True
-                print(f"target_fwd v. ref_fwd")
-                _print_diffs(target_fwd, ref_fwd)
+            print_debug_tensor_stats("target", target_fwd)
+            print_debug_tensor_stats("ref", ref_fwd)
+            print_debug_tensor_stats("diff", jnp.abs(target_fwd - ref_fwd))
```
Comment (zlsh80826) [inline tests/jax/test_distributed_fused_attn.py:553]: I’m curious if there are any issues with the old try-except pattern

**Review by zlsh80826 — APPROVED** [review]
LGTM

File: tests/jax/test_distributed_fused_attn.py (hunk 1 above)
Comment (mgoldfarb-nvidia) [reply tests/jax/test_distributed_fused_attn.py:553]: Brought this over from another PR where I did a lot more numeric debug. This feels a bit cleaner as the logic to enable printing is hidden and I can keep it on in both passing and failing cases.
